Pada bagian ini akan dilakukan fungsi inference untuk menguji model yang sudah dibuat sebelumnya

# Import Libraries

In [1]:
import pickle
import json
import pandas as pd
import numpy as np

# Load Files

In [2]:
with open('best_pipeline.pkl', 'rb') as file_1:
    model = pickle.load(file_1)

# Inferencing

In [3]:
data_inf = {'Hours_Studied': 2,
 'Attendance': 2,
 'Sleep_Hours': 6,
 'Previous_Scores': 8,
 'Tutoring_Sessions': 9,
 'Parental_Involvement': 'Medium',
 'Access_to_Resources': 'Medium',
 'Extracurricular_Activities': 'Yes',
 'Motivation_Level': 'Medium',
 'Family_Income': 'Medium',
 'Teacher_Quality': 'Medium',
 'School_Type': 'Public',
 'Peer_Influence': 'Positive',
 'Learning_Disabilities': 'No',
 'Parental_Education_Level': 'College',
 'Distance_from_Home': 'Near'}

data_inf = pd.DataFrame([data_inf])
data_inf

Hours_Studied  Attendance  Sleep_Hours  Previous_Scores  Tutoring_Sessions  \
0              2           2            6                8                  9   

  Parental_Involvement Access_to_Resources Extracurricular_Activities  \
0               Medium              Medium                        Yes   

  Motivation_Level Family_Income Teacher_Quality School_Type Peer_Influence  \
0           Medium        Medium          Medium      Public       Positive   

  Learning_Disabilities Parental_Education_Level Distance_from_Home  
0                    No                  College               Near

In [4]:
#Predict
pred = model.predict(data_inf)
pred

array([47.60957485])

In [5]:
data_inf['Exam_Score'] = pred
data_inf


Hours_Studied  Attendance  Sleep_Hours  Previous_Scores  Tutoring_Sessions  \
0              2           2            6                8                  9   

  Parental_Involvement Access_to_Resources Extracurricular_Activities  \
0               Medium              Medium                        Yes   

  Motivation_Level Family_Income Teacher_Quality School_Type Peer_Influence  \
0           Medium        Medium          Medium      Public       Positive   

  Learning_Disabilities Parental_Education_Level Distance_from_Home  \
0                    No                  College               Near   

   Exam_Score  
0   47.609575

Dari hasil prediksi di atas, siswa mendapatkan nilai 47.61 berdasarkan nilai-nilai yang diinput